specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor/preview/2020-01-13-preview/Microsoft.WorkloadMonitor.json

```diff
@@ -0,0 +1,805 @@
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "filter": {
+      "name": "$filter",
+      "in": "query",
+      "description": "list example: $filter=monitorName eq 'logical-disks|C:|disk-free-space-mb'; history example: $filter=isHeartbeat eq false",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "expand": {
+      "name": "$expand",
+      "in": "query",
+      "description": "ex: $expand=evidence,configuration",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "MonitorStateChangeList": {
+      "title": "Monitor History",
+      "description": "The monitor history of a monitor",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "value": {
+          "title": "State Change Array",
+          "description": "Array of state change transitions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MonitorStateChange"
+          }
+        },
+        "nextLink": {
+          "title": "Pageable next link",
+          "description": "Link to next page if list is too long.",
+          "type": "string"
+        }
+      }
+    },
+    "MonitorList": {
+      "title": "Monitor List",
+      "description": "Basic information about the current status of a monitor.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "value": {
+          "title": "Monitor Array",
+          "description": "Array of monitors.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Monitor"
+          }
+        },
+        "nextLink": {
+          "title": "Pageable next link",
+          "description": "Link to next page if list is too long.",
+          "type": "string"
+        }
+      }
+    },
+    "Monitor": {
+      "title": "Monitor Status",
+      "description": "Information about a monitor.",
+      "type": "object",
+      "readOnly": true,
+      "required": [
+        "id",
+        "name",
+        "type",
+        "properties"
+      ],
+      "properties": {
+        "id": {
+          "title": "Arm ID",
+          "description": "Arm ID of this monitor.",
+          "type": "string",
+          "example": "/subscriptions/bc27da3b-3ba2-4e00-a6ec-1fde64aa1e21/resourceGroups/tugamidiAlerts/providers/Microsoft.Compute/virtualMachines/linuxEUS/providers/Microsoft.WorkloadMonitor/monitors/logical-disks|C@3A"
+        },
+        "name": {
+          "title": "Url-encoded monitor name",
+          "description": "Url-encoded monitor name.",
+          "type": "string",
+          "example": "logical-disks|C@3A"
+        },
+        "type": {
+          "title": "Resource type",
+          "description": "Type of ARM resource.",
+          "type": "string",
+          "example": "Microsoft.WorkloadMonitor/monitors"
+        },
+        "properties": {
```

Comment:
The suggested modelling of this is to have a specific schema definition that defines the properties property, rether than an inline definition.  See:  as an example: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json#L1538

Reply:
I think I followed this example by defining an additional monitorProperties and a monitorStateChangeProperties